Automattic/newspack-plugin
# NRH Integration

### All Submissions:






### Changes proposed in this Pull Request:

This adds a "first class" [News Revenue Hub](https://fundjournalism.org/) integration in the Reader Revenue wizard. For sites using NRH checkout, we install [Newspack NRH Checkout](https://github.com/Automattic/newspack-nrh-checkout), which redirects the Donate block's form submission to NRH checkout with the appropriate post data. For this flow to work, these sites must have all of the Newspack donation plugins activated (WooCommerce, WooCommerce Subscriptions, WooCommerce Stripe Gateway, WooCommerce Name Your Price), which places an unnecessary burden on these sites since none of these are actually used. In addition to a cleaner setup for NRH customers, this branch allows these sites to operate without these plugins. In an upcoming PR I'm going to add support for cross-domain tracking so that NRH customers can capture analytics events on their donation page, and the UI work in this PR creates a logical home for this feature. Finally, this PR paves the way for integrations of other reader revenue tools that are in use on some Newspack sites.

On the first screen of the wizard a new select allows choosing between Newspack (WooCommerce) and News Revenue Hub approaches. If "News Revenue Hub" is selected, a few specific fields will be shown and none of the WC-dependent features will be. If "Newspack" is selected the flow will be roughly the same as it has been all along, with the exception of plugin installation. Originally, all of the WooCommerce-related plugins were required to use this wizard. Now only Newspack Blocks is required, and the WC ones are installed only after selecting Newspack.

There are also a few changes to the Donate block. If NRH Checkout is selected and the WooCommerce plugins are disabled, price tiers in the block must be configured manually. The "Configure Manually" toggle is hidden in this scenario and forced to be on. The only feature which cannot yet be configured manually is the currency symbol. Since all Newspack/NRH customers are U.S. based, I hard-coded the dollar sign. In the future we can make this selectable.

Test along with https://github.com/Automattic/newspack-blocks/pull/642

Closes https://github.com/Automattic/newspack-blocks/issues/623, https://github.com/Automattic/newspack-blocks/issues/518

### How to test the changes in this Pull Request:

To test you will need a News Revenue Hub client ID. You can get one by contacting me or from `Settings->Newspack NRH Checkout` on any NRH-enabled site. The key to testing this is ensuring that checkout works correctly in all five of these scenarios:

1. Regression one: Normal Newspack approach, after switching to this branch and doing nothing in the Reader Revenue wizard.
2. Regression two: NRH checkout using Newspack NRH Checkout after switching to this branch and doing nothing in the Reader Revenue wizard.
3. With all of the WooCommerce plugins disabled, go to Reader Revenue wizard, select Newspack and go through the setup steps. Ideally, start by resetting all Newspack options, which can be done in Settings->Newspack by checking "Reset Newspack" and clicking "Save Changes."
4. Remove the Newspack NRH Checkout plugin. After setting up WooCommerce donations, switch to NRH Checkout, and ensure it works properly. 
5. Disable WooCommerce plugins. You will need to set price tiers in the Donate block manually, then verify everything is still working.

If NRH checkout is selected and WooCommerce disabled, the Donate block will only appear if configured manually. 

### Other information:
